compiler/luci/service/src/CircleShapeInferenceRule.cpp

```diff
@@ -1494,6 +1494,13 @@ class ShapeInferenceAlgorithm final : public luci::CircleNodeVisitor<loco::NodeS
     return loco::NodeShape{output_shape};
   }
 
+  loco::NodeShape visit(const luci::CircleSegmentSum *node) final
+  {
+    auto input_shape = loco::shape_get(node->input()).as<loco::TensorShape>();
```

Comment:
This is not how this operator works
You can determine output shape either from built-in shape in tflite/circle model or from maximum number (last index) in segmen_ids input.